implementations/code_df/metric.py

```diff
@@ -33,3 +34,45 @@ def _flatten(self, item):
         """
 
         raise NotImplementedError
+
+    def _aggregate(self, series):
+        """
+        Aggregate the re-sampled DataFrame
+        """
+
+        raise NotImplementedError
+
+    def compute_time_series(self, cols=None, period='M'):
```

Comment:
`time_series` could be a shorter, and still meaningful, name.

Reply:
Will do!